client/src/components/App.jsx

```diff
@@ -37,15 +38,43 @@ class App extends Component {
     return this.location;
   }
 
+  getShippingCostData(costArray, id) {
+    for (let i = 0; i < costArray.length; i++) {
+      if (id === costArray[i].product_id) {
+        return costArray[i];
+      }
+    }
+  }
+
+  getShippingCost(data) {
+    if (data.isFreeShipping) {
+      return 'Free';
+    } else {
+      let cost = '$' + data.costOfDelivery.toString();
+      return cost;
+    }
+  }
+
   async getProduct() {
     const id = this.setProductId();
     const location = this.setProductLocation();
     try {
       const response = await axios.get(`${location}/shipping-api/${id}`);
       const shipping = response.data;
+      const costResponse = await axios.get('https://valeriia-ten-inventory.s3.us-east-2.amazonaws.com/100inventory.json');
+      const itemShippingCostData = this.getShippingCostData(costResponse.data, id);
```

Comment:
You could do these 2 requests in parallel with Promise.all to speed up load time.